tests/shared/sdk/usbboot.spec.js

```diff
@@ -0,0 +1,160 @@
+        this.usbGetDeviceListStub = m.sinon.stub(usb, 'getDeviceList')
+        this.usbGetDeviceListStub.returns([])
+      })
+
+      afterEach(function () {
+        this.usbGetDeviceListStub.restore()
+      })
+
+      it('should resolve an empty array', function () {
+        return usbboot.scan().then((devices) => {
+          m.chai.expect(devices).to.deep.equal([])
+        })
+      })
+    })
+
+    describe('given a non supported device', function () {
+      beforeEach(function () {
+        this.usbGetDeviceListStub = m.sinon.stub(usb, 'getDeviceList')
+        this.usbGetDeviceListStub.returns([
+          createUSBDeviceStub({
+            vendorID: 0x9999,
+            productID: 0x2763,
+            manufacturerIndex: 1,
+            productIndex: 2,
+            interfaces: 2,
+            strings: [
+              null,
+              'FOO',
+              'X2763'
+            ]
+          })
+        ])
+      })
+
+      afterEach(function () {
+        this.usbGetDeviceListStub.restore()
+      })
+
+      it('should resolve an empty array', function () {
+        return usbboot.scan().then((devices) => {
+          m.chai.expect(devices).to.deep.equal([])
+        })
+      })
+    })
+
+    describe('given a supported and a non supported device', function () {
+      beforeEach(function () {
+        this.usbGetDeviceListStub = m.sinon.stub(usb, 'getDeviceList')
+        this.usbGetDeviceListStub.returns([
+          BCM2835_STUB,
+          createUSBDeviceStub({
+            vendorID: 9999,
+            productID: 9999,
+            manufacturerIndex: 1,
+            productIndex: 2,
+            interfaces: 1,
+            strings: [
+              null,
+              'Foo',
+              'Bar'
+            ]
+          })
+        ])
+      })
+
+      afterEach(function () {
+        this.usbGetDeviceListStub.restore()
+      })
+
+      it('should only resolve the supported device', function () {
+        return usbboot.scan().then((devices) => {
+          m.chai.expect(devices.length).to.equal(1)
+          const device = _.first(devices)
+          m.chai.expect(device.name).to.equal('Broadcom BCM2708 Boot')
+        })
+      })
+    })
+
+    describe('given a BCM2835 device', function () {
+      beforeEach(function () {
+        this.usbGetDeviceListStub = m.sinon.stub(usb, 'getDeviceList')
+        this.usbGetDeviceListStub.returns([ BCM2835_STUB ])
+      })
+
+      afterEach(function () {
+        this.usbGetDeviceListStub.restore()
+      })
+
+      it('should resolve a friendly device object', function () {
+        return usbboot.scan().then((devices) => {
+          m.chai.expect(devices.length).to.equal(1)
+          const device = _.first(devices)
+          m.chai.expect(device.name).to.equal('Broadcom BCM2708 Boot')
+          m.chai.expect(device.interface).to.equal(0)
+          m.chai.expect(device.outEndpoint).to.equal(1)
+          m.chai.expect(device.device).to.deep.equal(BCM2835_STUB)
```

Comment:
`device.device` looks a tiny bit strange here - might be worth giving the outer object a different name, just to clarify things?